src/parse/properties/css_property_parser_gen.c

```diff
@@ -660,19 +671,19 @@ int main(int argc, char **argv)
 			only_ident = false;
 		} else if (strcmp(rkv->key, "GENERIC") == 0) {
 			is_generic = true;
+			free(rkv); /* Free GENERIC token */
```

Comment:
Rather than all these `free(rkv)`, in almost every branch, I'd rather see a single `free(rkv)` at the bottom of the loop and in the few places we actually use `rkv` something like:

```c
FOO.item[FOO.count++] = rkv;
rkv = NULL; /* Ownership moved to FOO array */
```

Reply:
I have added a flag to indicate when rkv should be freed - this reduces new code needed